CMakeLists.txt

```diff
@@ -42,47 +79,81 @@ set (lodepng_src
 add_library(zopflilib_obj OBJECT
   ${zopflilib_src}
 )
-if(BUILD_SHARED_LIBS)
+if(ZOPFLI_BUILD_SHARED)
   set_property(TARGET zopflilib_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
 
 #
 # libzopfli
 #
-add_library(libzopfli
+add_library(libzopfli ${zopfli_library_type}
   $<TARGET_OBJECTS:zopflilib_obj>
 )
+target_include_directories(libzopfli INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/src/zopfli")
 set_target_properties(libzopfli PROPERTIES
   OUTPUT_NAME zopfli
-  VERSION 1.0.2
-  SOVERSION 1
+  VERSION ${ZOPFLI_VERSION}
+  SOVERSION ${ZOPFLI_VERSION_MAJOR}
 )
-if(UNIX)
-  target_link_libraries(libzopfli m)
+if(UNIX AND NOT (BEOS OR HAIKU))
+  target_link_libraries(libzopfli PRIVATE m)
 endif()
 
 #
 # libzopflipng
 #
-add_library(libzopflipng
+add_library(libzopflipng ${zopfli_library_type}
   ${zopflipnglib_src}
   ${lodepng_src}
   $<TARGET_OBJECTS:zopflilib_obj>
 )
+target_include_directories(libzopflipng INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/src/zopflipng")
 set_target_properties(libzopflipng PROPERTIES
   OUTPUT_NAME zopflipng
-  VERSION 1.0.2
-  SOVERSION 1
+  VERSION ${ZOPFLI_VERSION}
+  SOVERSION ${ZOPFLI_VERSION_MAJOR}
 )
 
+# MSVC does not export symbols by default when building a DLL, this is a
+# workaround for recent versions of CMake
+if(MSVC AND ZOPFLI_BUILD_SHARED)
+  if(CMAKE_VERSION VERSION_LESS 3.4)
+    message(WARNING "Automatic export of all symbols to DLL not supported until CMake 3.4")
+  else()
+    set_target_properties(libzopfli PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    set_target_properties(libzopflipng PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+  endif()
+endif()
+
 #
 # zopfli
 #
 add_executable(zopfli src/zopfli/zopfli_bin.c)
 target_link_libraries(zopfli libzopfli)
+if(MSVC)
+  target_compile_definitions(zopfli PRIVATE _CRT_SECURE_NO_WARNINGS)
+endif()
 
 #
 # zopflipng
 #
 add_executable(zopflipng src/zopflipng/zopflipng_bin.cc)
 target_link_libraries(zopflipng libzopflipng)
+if(MSVC)
+  target_compile_definitions(zopflipng PRIVATE _CRT_SECURE_NO_WARNINGS)
+endif()
+
+#
+# Install
+#
+if(ZOPFLI_BUILD_INSTALL)
+  include(GNUInstallDirs)
+  install(TARGETS libzopfli libzopflipng zopfli zopflipng
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  )
+  install(FILES include/zopfli.h include/zopflipng_lib.h
```

Comment:
That line seems to cause an install error:
```console
# make DESTDIR="$PWD/DESTDIR" install
[..]
CMake Error at cmake_install.cmake:171 (file):
  file INSTALL cannot find
  "/tmp/tmp.EK9KAgZMcc/jibsen-zopfli--update-cmake/include/zopfli.h".

# find -name zopfli.h
./src/zopfli/zopfli.h
```

Reply:
Ah, I copied the install section from an older version of the CMake script, which used the previous directory structure. I will update that, thanks!

Reply:
Works better now, thanks!

```
# make DESTDIR="$PWD/DESTDIR" install
[..]
# find DESTDIR/ -not -type d | sort
DESTDIR/usr/local/bin/zopfli
DESTDIR/usr/local/bin/zopflipng
DESTDIR/usr/local/include/zopfli.h
DESTDIR/usr/local/include/zopflipng_lib.h
DESTDIR/usr/local/lib64/libzopflipng.so
DESTDIR/usr/local/lib64/libzopflipng.so.1
DESTDIR/usr/local/lib64/libzopflipng.so.1.0.2
DESTDIR/usr/local/lib64/libzopfli.so
DESTDIR/usr/local/lib64/libzopfli.so.1
DESTDIR/usr/local/lib64/libzopfli.so.1.0.2
```